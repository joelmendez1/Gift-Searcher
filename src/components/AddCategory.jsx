import { useState } from 'react';
import PropTypes from "prop-types"

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = ({ target }) => {
    setInputValue(target.value);
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    const newInputValue = inputValue.trim();

    if(newInputValue.length <= 1) return;

    onNewCategory(newInputValue);
    setInputValue('');
  }

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input 
      type='text' 
      placeholder='Buscar gifs'
      value={inputValue}
      onChange={handleInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
