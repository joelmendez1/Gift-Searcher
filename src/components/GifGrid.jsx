import React, { useEffect, useState } from 'react'
import { getGifs } from '../utils/getGifs'
import { GifItem } from './GifItem';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs(category)
      .then(res => setImages(res))
  }, [category])

  return (
    <>
      <h3>{category}</h3>
      <div className='card-grid'>
        {
          images.map((image) => {
            return <GifItem key={image.id} {...image}/>
          })
        }
      </div>
    </>
  )
}
