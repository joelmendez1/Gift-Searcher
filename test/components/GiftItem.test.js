import { render } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Test in GifItem component', () => {
  test('Should match snapshot', () => {
    const title = "Titulo";
    const url = "https://one-punch.com/saitama.jpg";
    const { container } = render
      (
        <GifItem title={title} url={url} />
      )

    expect(container).toMatchSnapshot();
  })
})