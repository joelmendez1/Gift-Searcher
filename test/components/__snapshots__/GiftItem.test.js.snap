// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifItem component Should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Titulo
    </p>
  </div>
</div>
`;
